Classes/Backend/TableConfigurationPostProcessor.php

```diff
@@ -49,11 +54,114 @@ class TableConfigurationPostProcessor implements TableConfigurationPostProcessin
+                    'mod.wizards.newContentElement.wizardItems.fluxContent {
+                    header = LLL:EXT:flux/Resources/Private/Language/locallang.xlf:newContentWizard.fluxContent
+                    show = *
+                    elements {
+                    }'
+                );
+                $initialized = true;
+            }
+
+            // Registration for "new content element" wizard to show our new CType (otherwise, only selectable via "Content type" drop-down)
+            ExtensionManagementUtility::addPageTSConfig(
+                sprintf(
+                    'mod.wizards.newContentElement.wizardItems.fluxContent.elements.%s {
+                    iconIdentifier = %s
+                    title = LLL:EXT:%s/Resources/Private/Language/locallang.xlf:flux.%s
+                    description = LLL:EXT:%s/Resources/Private/Language/locallang.xlf:flux.%s.description
+                    tt_content_defValues {
+                        CType = %s
+                    }
+                }',
+                    $formId,
+                    $iconIdentifier,
+                    $extensionKey,
+                    $formId,
+                    $extensionKey,
+                    $formId,
+                    $contentType
+                )
+            );
+
+            // Create the TCA necessary to drive this custom CType
+            $showItem = '
+                --palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:palette.general;general,
+                --palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:palette.header;header,pi_flexform,
+                --div--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:tabs.appearance,layout;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:layout_formlabel,
+                --palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:palette.appearanceLinks;appearanceLinks,
+                --div--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:tabs.access,hidden;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:field.default.hidden,
+                --palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:palette.access;access,
+                --div--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:tabs.extended,rowDescription,
+                --div--;LLL:EXT:lang/locallang_tca.xlf:sys_category.tabs.category,categories, 
+                --div--;LLL:EXT:flux/Resources/Private/Language/locallang.xlf:tt_content.tabs.relation, tx_flux_parent, tx_flux_column, tx_flux_children
+            ';
+            $GLOBALS['TCA']['tt_content']['types'][$contentType]['showitem'] = $showItem;
+            $GLOBALS['TCA']['tt_content']['columns']['pi_flexform']['ds']['*,' . $contentType] = [];
+            ExtensionManagementUtility::addToAllTCAtypes('tt_content', 'pi_flexform', $contentType);
+        }
+    }
+
+    /**
```

Comment:
imho this method is a bit large, could refactored into a few smaller ones which would improve readability